src/auth/digest/eDirectory/edir_ldapext.cc

```diff
@@ -399,7 +399,7 @@ static int nmasldap_get_simple_pwd(
     }
 
     if (pwdBuf != NULL) {
-        memset(pwdBuf, 0, bufferLen);
+        memset_s(pwdBuf, bufferLen, 0, bufferLen);
```

Comment:
The combination of this usage and memset_s() replacement implementation contradicts the PR intent AFAICT: 
Calling memset_s() instead of memset() implies that the caller thinks that it is important that the buffer is cleared. Ignoring memset_s() errors implies that the caller relies on the [constraint handler](https://en.cppreference.com/w/c/error/set_constraint_handler_s) to handle clearing errors, but the replacement never calls any constraint handlers, relying on the caller to check for them instead! Moreover, Squid never sets the constraint handler, resulting in essentially undefined behavior when the standard memset_s() is used!

IIRC, this contradiction is closely related to my other, old concern:  Standard memset_s() calls the [constraint handler](https://en.cppreference.com/w/c/error/set_constraint_handler_s). A developer adding a memset_s() call to their code has the right to expect that handler to be called (instead of checking every memset_s() call result). By completely ignoring that critical part of the standard memset_s() API, this PR needlessly increases the uncertainty of the outcome on caller errors:

* When standard memset_s() is available, errors will
    * be always silently ignored in some environments,
    * always crash Squid (without Squid-specific logging/diagnostics) in some environments, and
    * possibly do something else in other environments.
* When standard memset_s() is unavailable, errors will
    * be always silently ignored in all environments.

A PR that focuses on improving Squid security ought to do a lot better than introduce such a variety of bad outcomes when that security is threatened!

There are several ways to fix this problem. I suggest the simplest fix I can think of: Instead of trying to replace a very-difficult-to-correctly-replace memset_s(), provide a high-level equivalent. We can call it Security::ZeroSensitiveMemory() or some such. The function should check for all the caller errors a memset_s() would check, use the volatile pointer trick to call memset(), and return `void`. The function should throw on errors or, if others prefer that, assert() or abort(). 

With the fix proposed above, there is no need to check for memset_s() support or to call memset_s() if it is supported!